load-gen/src/test/java/io/smartcat/data/loader/util/LinkedEvictingBlockingQueueTest.java

```diff
@@ -0,0 +1,102 @@
+package io.smartcat.data.loader.util;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+public class LinkedEvictingBlockingQueueTest {
+
+    @Test(timeout = 1000)
+    public void take_should_block_thread_until_element_is_available_in_queue() throws InterruptedException {
+        // GIVEN
+        LinkedEvictingBlockingQueue<Integer> queue = new LinkedEvictingBlockingQueue<>();
+        AtomicBoolean setFromPutInQueueThread = new AtomicBoolean(false);
+        Thread putInQueueThread = new Thread(() -> {
+            try {
+                Thread.sleep(500);
+            } catch (InterruptedException e) {
+            }
+            setFromPutInQueueThread.set(true);
+            queue.put(2);
+        });
+        putInQueueThread.setDaemon(true);
+
+        // WHEN
+        putInQueueThread.start();
+        int val = queue.take();
+
+        // THEN
+        Assert.assertEquals(2, val);
+        Assert.assertTrue(setFromPutInQueueThread.get());
+    }
+
+    @Test(timeout = 6000)
+    public void should_not_take_value_from_queue_when_value_is_not_put() throws InterruptedException {
+        // GIVEN
+        LinkedEvictingBlockingQueue<Integer> queue = new LinkedEvictingBlockingQueue<>();
+        Thread putInQueueThread = new Thread(() -> {
+            try {
+                Thread.sleep(5000);
```

Comment:
Consider implementing unit tests without `Thread.sleep()` and replace them with sync objects (e.g. latches or semaphores).